### PR TITLE
Support for parametric curves/surfaces/volumes

### PR DESCRIPTION
**Pull request by maltezfaria**
The goal of this `PR` is to implement some basic functionality so that we can deal with simple parametric shapes natively. This requires:
- [x] Extend `GeometricEntity` to possibly store an explicit parametrisation when available.
- [x] Ability to create a `ParametricElement`, which is is geometric in that it stores a general function mapping the reference domain t o the element itself (and not e.g. a polynomial or the nodes specifying a polynomial interpolant)
- [x] Functionality to mesh a `GeometricEntity` if it has a valid parametrization

### CODE REVIEW

File: test/mesh_test.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    gmsh.option.setNumber("General.Verbosity", 2)
+    gmsh.option.setNumber("Mesh.MeshSizeMax", 0.005)
+    Inti.clear_entities!()
+    gmsh.model.occ.addDisk(0, 0, 0, rx, ry)
+    gmsh.model.occ.synchronize()
+    gmsh.model.mesh.generate(2)
+    Ω, M = Inti.import_mesh_from_gmsh_model(; dim = 2)
+    gmsh.finalize()
+    Γ = Inti.external_boundary(Ω)
+    dict = Inti.topological_neighbors(M[Γ])
```
Comment (tanderson92) [inline test/mesh_test.jl:49]: This does not seem to have been implemented, or rather pushed to this PR.

Comment (maltezfaria) [reply test/mesh_test.jl:49]: You are right, this is in some other brach related to the local dim methods. Do
you want it or can I leave it to be added later?

Comment (maltezfaria) [reply test/mesh_test.jl:49]: But you spotted a problem: the `runtests.jl` did not run all tests. I will fix
it now.

Comment (tanderson92) [reply test/mesh_test.jl]: It doesn't seem essential to have it in this PR, and since it is logically distinct it can be added a bit later.

**Comment by tanderson92** [comment]
Am I supposed to be able to `viz()` ParametricElements? I thought I had seen you be able to do that. With the msh that results from "Native mesh generation" I get, with `viz(collect(Inti.elements(msh))[1])`:

> julia> viz(collect(Inti.elements(msh))[1])
ERROR: UndefVarError: `to` not defined
Stacktrace:
  [1] (::MeshesMakieExt.var"#211#212")(p::Meshes.Point2)
    @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:217
  [2] iterate(g::Base.Generator, s::Vararg{Any})
    @ Base ./generator.jl:47 [inlined]
  [3] _collect(c::Vector{…}, itr::Base.Generator{…}, ::Base.EltypeUnknown, isz::Base.HasShape{…})
    @ Base ./array.jl:854
  [4] collect_similar
    @ ./array.jl:763 [inlined]
  [5] map
    @ ./abstractarray.jl:3282 [inlined]
  [6] segmentsof(topo::GridTopology{1}, vert::Vector{Meshes.Point2})
    @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:217
  [7] #152
    @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:34 [inlined]
  [8] map(::MeshesMakieExt.var"#152#157", ::Observable{SimpleMesh{…}}; ignore_equal_values::Bool, priority::Int64)
    @ Observables ~/.julia/packages/Observables/YdEbO/src/Observables.jl:570
  [9] map
    @ ~/.julia/packages/Observables/YdEbO/src/Observables.jl:568 [inlined]
 [10] vizmesh1D!(plot::Plot{Meshes.viz, Tuple{SimpleMesh{2, Float64, Vector{Meshes.Point2}, GridTopology{1}}}})
    @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:31

**Comment by maltezfaria** [comment]
> Am I supposed to be able to `viz()` ParametricElements? I thought I had seen you be able to do that. With the msh that results from "Native mesh generation" I get, with `viz(collect(Inti.elements(msh))[1])`:
> 
> > julia> viz(collect(Inti.elements(msh))[1])
> > ERROR: UndefVarError: `to` not defined
> > Stacktrace:
> > [1] (::MeshesMakieExt.var"#211#212")(p::Meshes.Point2)
> > @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:217
> > [2] iterate(g::Base.Generator, s::Vararg{Any})
> > @ Base ./generator.jl:47 [inlined]
> > [3] _collect(c::Vector{…}, itr::Base.Generator{…}, ::Base.EltypeUnknown, isz::Base.HasShape{…})
> > @ Base ./array.jl:854
> > [4] collect_similar
> > @ ./array.jl:763 [inlined]
> > [5] map
> > @ ./abstractarray.jl:3282 [inlined]
> > [6] segmentsof(topo::GridTopology{1}, vert::Vector{Meshes.Point2})
> > @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:217
> > [7] #152
> > @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:34 [inlined]
> > [8] map(::MeshesMakieExt.var"#152#157", ::Observable{SimpleMesh{…}}; ignore_equal_values::Bool, priority::Int64)
> > @ Observables ~/.julia/packages/Observables/YdEbO/src/Observables.jl:570
> > [9] map
> > @ ~/.julia/packages/Observables/YdEbO/src/Observables.jl:568 [inlined]
> > [10] vizmesh1D!(plot::Plot{Meshes.viz, Tuple{SimpleMesh{2, Float64, Vector{Meshes.Point2}, GridTopology{1}}}})
> > @ MeshesMakieExt ~/.julia/packages/Meshes/GKlCN/ext/simplemesh.jl:31

This worked on my end, but trying to visualize a vector of elements did not. The issue was that `ParametricElement` somehow was *not* a subtype of `ReferenceInterpolant`. I will fix that.

**Review by tanderson92 — APPROVED** [review]
LGTM

**Pull request merged**